test/errors/ifBranches-detailed.good

```diff
@@ -0,0 +1,21 @@
+─── error in ifBranches.chpl:2 [IncompatibleIfBranches] ───
+  Branches of if-expression have incompatible types.
+  In the following if-expression:
+      |
+    2 | var y = if b then 42 else "hello";
+      |                   ⎺⎺       ⎺⎺⎺⎺⎺⎺  
+      |
+  the first branch is a param of type 'int(64)', while the second is a param of type 'string'.
+
+─── error in ifBranches.chpl:5 [IncompatibleIfBranches] ───
+  Branches of if-expression have incompatible types.
+  In the following if-expression:
+      |
+    5 |   if b
+    6 |   then 42
+      |        ⎺⎺ 
+    7 |   else "hello";
+      |         ⎺⎺⎺⎺⎺⎺  
+      |
+  the first branch is a param of type 'int(64)', while the second is a param of type 'string'.
```

Comment:
The mentions of `param` here, while very precise, also don't really matter—which is to say, if these were not params, but vars or consts, the error would still hold.  For that reason, I'd probably just say "is of type..." and "is of type..." to avoid any incorrect assumption that "if only they hadn't been params, this wouldn't have been an error...?"  Also, I think the word 'param' in general isn't very intuitive for users when describing something that simply appears to be a literal value.